### PR TITLE
Move electron app in packages/ dir

### PR DESCRIPTION
**Pull request by lgeiger**
Make the electron app a proper lerna package 📦 

This is a initial *working* version. Feel free to commit to this branch for further fixes.

Dependencies aren't hoisted anymore. With hoisting enabled build dependencies weren't available when running npm commands inside a lerna package folder. Only hoisted dependency is `RxJS` because otherwise the Electron app won't start.

## Todo: *(help appreciated)*
- [x] Fix package tests (Why isn't [this mock](https://github.com/nteract/nteract/compare/full-lerna?expand=1#diff-1e8bdd08b92797c747f1525daafd30ddR15) working anymore)
- [x] Fix desktop app tests
- [x] Cleanup extraneous dependencies
- [x] Document `npm` commands

### CODE REVIEW

**Comment by rgbkrk** [comment]
I _think_ it's ok to hoist still. The annoying thing is after you do this, you'll probably have to `git clean -xfd`.

**Comment by lgeiger** [comment]
> I think it's ok to hoist still

It is, but in my experience it makes it hard to ensure that all dependencies are included in a package. E.g. `transform-vega` requires `d3` for webpack to do it's thing but it was never a issue with hoisting because `d3` is a direct dependency of another transform anyway.
Apart from that installing packages with `npm` inside a package folder also can mess a things up in strange ways. So I opted for not hoisting packages for now.

**Comment by lgeiger** [comment]
Do we really need to run `flow-copy-source` in the `prepare` script during bootstrap? We only need them when publishing the packages, right?

`flow-copy-source` takes a really long time to do it's job.

**Comment by rgbkrk** [comment]
> Do we really need to run flow-copy-source in the prepare script during bootstrap? We only need them when publishing the packages, right?

They also provide flow types during development, so while not frequent if they're out of sync you'll get weird flow errors.

**Comment by lgeiger** [comment]
This is ready for people to try out and see if it works on their machines.

File: README.md [hunk 1]
```diff
@@ -102,7 +101,7 @@ And then opening `docs/index.html` in your favorite browser.
 
 > I upgraded my developer installation and things are broken!
 
-- Try `npm run reinstall`
+- Try `git clean -xdf && npm i`
```
Comment (rgbkrk) [inline README.md:104]: 😅

**Comment by rgbkrk** [comment]
Alright, I'll give it a go!

**Comment by rgbkrk** [comment]
I love seeing all these references to the packages by name rather than relative path.

**Comment by rgbkrk** [comment]
I see now rxjs and uuid are hoisted. What made it so you had to hoist uuid?

**Comment by rgbkrk** [comment]
Took 133s to install from scratch (I did the `git clean -xfd` first).

**Comment by rgbkrk** [comment]
## Bugs encountered

* none of the examples load. It looks like they're one path too far.

screen shot 2017-09-14 at 8 31 56 am

## Extra things I tested:

* Built in node.js kernel is working like a champ
* Various renderers (vega, plotly) are all working

**Comment by rgbkrk** [comment]
One extra thing -- running the command line tools install created `packages/desktop/bin/nteract-env`. Maybe that's normal behavior? Maybe it's just something to gitignore?

**Comment by rgbkrk** [comment]
Fixed the one issue I saw, now to see how changing other packages goes.

**Comment by rgbkrk** [comment]
Working like a champ, webpack is correctly building from our `src` folder in respective packages.

The only weird thing is that `lerna` masks the current status of webpack:

```
$ npm run start

> nteract@0.2.0 start /Users/kylek/code/src/github.com/nteract/nteract-maint
> lerna run start --scope @nteract/desktop

lerna info version 2.1.2
lerna info versioning independent
lerna info scope @nteract/desktop
⸨░░░░░░░░░░░░░░░░░░⸩ ⠹ run: info scope @nteract/desktop
```

**Comment by rgbkrk** [comment]
This is AWESOME. Thank you @lgeiger!

🎉

**Comment by rgbkrk** [comment]
The only other script I'm seeing as needing to move is `scripts/make_icons.sh`. Then again I haven't _used_ that script in a while.

**Comment by rgbkrk** [comment]
Guess I should test by making a fake release too.

**Comment by rgbkrk** [comment]
```
~/code/src/github.com/nteract/nteract-maint/packages/desktop$ npm run dist

> @nteract/desktop@0.2.0 dist /Users/kylek/code/src/github.com/nteract/nteract-maint/packages/desktop
> npm run clean && npm run build:webpack && electron-builder

npm ERR! missing script: clean

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/kylek/.npm/_logs/2017-09-14T15_47_18_636Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @nteract/desktop@0.2.0 dist: `npm run clean && npm run build:webpack && electron-builder`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @nteract/desktop@0.2.0 dist script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/kylek/.npm/_logs/2017-09-14T15_47_18_664Z-debug.log
```

**Comment by rgbkrk** [comment]
Not that I expected this to work but hey:

```
$ lerna run dist --scope @nteract/desktop
lerna info version 2.0.0
lerna info versioning independent
lerna info scope @nteract/desktop
lerna ERR! dist Errored while running script in '@nteract/desktop'
lerna ERR! execute Error: Command failed: npm run dist
lerna ERR! execute npm ERR! missing script: clean
lerna ERR! execute
lerna ERR! execute npm ERR! A complete log of this run can be found in:
lerna ERR! execute npm ERR!     /Users/kylek/.npm/_logs/2017-09-14T15_47_57_997Z-debug.log
lerna ERR! execute npm ERR! code ELIFECYCLE
lerna ERR! execute npm ERR! errno 1
lerna ERR! execute npm ERR! @nteract/desktop@0.2.0 dist: `npm run clean && npm run build:webpack && electron-builder`
lerna ERR! execute npm ERR! Exit status 1
lerna ERR! execute npm ERR!
lerna ERR! execute npm ERR! Failed at the @nteract/desktop@0.2.0 dist script.
lerna ERR! execute npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
lerna ERR! execute
lerna ERR! execute npm ERR! A complete log of this run can be found in:
lerna ERR! execute npm ERR!     /Users/kylek/.npm/_logs/2017-09-14T15_47_58_024Z-debug.log
lerna ERR! execute
lerna ERR! execute > @nteract/desktop@0.2.0 dist /Users/kylek/code/src/github.com/nteract/nteract-maint/packages/desktop
lerna ERR! execute > npm run clean && npm run build:webpack && electron-builder
lerna ERR! execute
```

**Comment by lgeiger** [comment]
Thanks for the review!

> I see now rxjs and uuid are hoisted. What made it so you had to hoist uuid?

[This mock](https://github.com/nteract/nteract/pull/1891/files#diff-1e8bdd08b92797c747f1525daafd30ddR15) didn't work otherwise.
I'm happy to hoist more things but if we hoist things like `jmp`, we'd need to modify the `electron-builder` setup.

> One extra thing -- running the command line tools install created packages/desktop/bin/nteract-env. Maybe that's normal behavior? Maybe it's just something to gitignore?

Yep that's expected. We need to modify the gitingore.

> The only weird thing is that lerna masks the current status of webpack:

Adding `--stream` to the command should fix it.

> Guess I should test by making a fake release too.

That would be great! I can fix the other two issues you mentioned.

**Comment by rgbkrk** [comment]
> That would be great! I can fix the other two issues you mentioned.

Cool! Yeah I haven't figured out the dist issue, even after doing the pieces here.

**Comment by lgeiger** [comment]
> Yeah I haven't figured out the dist issue, even after doing the pieces here.

OK, running the dist commands from the root seem to work. I'll push a fix soon.

**Comment by lgeiger** [comment]
Well hoisting `uuid` break the packaged app.

Do you know a way to fix [this test](https://github.com/nteract/nteract/pull/1891/files#diff-1e8bdd08b92797c747f1525daafd30ddR15) without hoisting?

**Comment by rgbkrk** [comment]
I'm going to have to play with the uuid one there, it's weird that it matters that it has to be hoisted. I wonder if it's because we don't add it into the webpack build.

**Comment by lgeiger** [comment]
I was able to fix the mocking by setting `moduleNameMapper` in the jest config. Let's see what Travis thinks.

**Comment by rgbkrk** [comment]
Hey! Travis is passing. Should I give this another spin?

**Comment by lgeiger** [comment]
> Should I give this another spin?

Yeah would be great.

**Comment by rgbkrk** [comment]
There's some "prettier churn". I'll go ahead and upgrade and do a `npm run prettier`. Expect at least one commit. 😜

**Comment by rgbkrk** [comment]
Oooooh the latest lerna has beautiful outputs

**Comment by rgbkrk** [comment]
Examples are working well!

Running the `dist` is currently in progress 🤞

**Comment by rgbkrk** [comment]
The `dist` is very successful. I'd say we're fit to merge this after CI passes!

**Comment by rgbkrk** [comment]
image

**Pull request merged**